File: elixir/raindrops/raindrops.exs
defmodule Raindrops do

  @raindrops %{3 => "Pling", 5 => "Plang", 7 => "Plong"}

  @spec convert(pos_integer) :: String.t()
  def convert(number) do
    case modulo(number, Map.keys(@raindrops)) do
      "" -> "#{number}"
      raindrops_sound -> raindrops_sound
    end
  end

  defp modulo(_, []),
    do: ""
  defp modulo(number, [divisor | rest]) when rem(number, divisor) == 0,
    do: @raindrops[divisor] <> modulo(number, rest)
  defp modulo(number, [_ | rest]),
    do: "" <> modulo(number, rest)
end

File: c/raindrops/src/raindrops.c
#include "stdio.h"
#include "string.h"
#include "raindrops.h"

void convert(char* result, int drops)
{
  if (drops % 3 == 0) strcat(result, "Pling");
  if (drops % 5 == 0) strcat(result, "Plang");
  if (drops % 7 == 0) strcat(result, "Plong");

  if (strlen(result) == 0) sprintf(result, "%d", drops);
}
